proof/invariant-abstract/DetSchedSchedule_AI.thy

```diff
@@ -22536,21 +22394,26 @@ lemma call_kernel_valid_sched:
                   apply (clarsimp simp: obj_at_def vs_all_heap_simps active_sc_def)
                  apply (clarsimp simp: consumed_time_bounded_def current_time_bounded_def)
                 apply (clarsimp simp: obj_at_def vs_all_heap_simps active_sc_def)
+                apply (clarsimp simp: current_time_bounded_def consumed_time_bounded_def)
+                apply (clarsimp simp: obj_at_def vs_all_heap_simps active_sc_def
+                                      cur_sc_more_than_ready_def refill_ready_no_overflow_def)
                apply (erule current_time_bounded_strengthen, simp)
               apply (erule ct_in_state_weaken, case_tac st; simp)
              apply (erule ct_in_state_weaken, case_tac st; simp add: is_BlockedOnReceive_def)
             apply (erule current_time_bounded_strengthen, simp)
+           apply (clarsimp simp: current_time_bounded_def consumed_time_bounded_def)
            apply (erule ct_in_state_weaken, case_tac st; simp)
           apply (erule ct_in_state_weaken, case_tac st; simp add: is_BlockedOnReceive_def)
          apply (erule cur_sc_chargeable_cur_sc_in_release_q_imp_zero_consumed; simp)
         apply (clarsimp simp: obj_at_def vs_all_heap_simps active_sc_def cur_sc_more_than_ready_def)
        apply (clarsimp simp: ct_ready_if_schedulable_def)
       apply (fastforce simp: heap_refs_inv_def intro: invs_retract_sc_tcbs invs_retract_tcb_scps)
      apply wpsimp
+    apply wpsimp
     apply (rule_tac Q="\<lambda>_. valid_sched
                            and invs
                            and consumed_time_bounded
-                           and next_time_bounded 2
+                           and next_time_bounded 3
```

Comment:
I'm surprised that this is needed. Perhaps double check this?

But there is no problem if this needs to be larger, as long as it's constant and small.

Reply:
Although now that I think about it, it's not surprising if changes to refill logic would affect this.

Reply:
The new version of `refill_update` will definitely require us to increase the buffer we have before `max_time`.  I'm not 100% sure that `refill_budget_check` will as well, but `refill_update` needing this is enough to make it necessary globally